frontend/src/components/GifWidget/GifWidget.tsx

```diff
@@ -0,0 +1,94 @@
+import { SetStateAction, useEffect, useState } from "react";
+import {
+    GifWidgetStyled,
+    GifWidgetSearch,
+    GifList,
+    GifListImage,
+    GifWidgetAtrtibutionSection,
+    GiphyAttributionLogo
+} from "./GifWidget.styled";
+import axios from "axios";
+import GiphyAttributionLogoImage from "assets/GiphyAttributionLogo.png";
+
+interface IGifWidget {
+    id: string;
+    images: Record<
+        string,
+        {
+            url: string;
+        }
+    >;
+}
+
+const GifWidget = () => {
+    const [gifsLoading, setGifsLoading] = useState(true);
+    const [gifList, setGifList] = useState<IGifWidget[]>([]);
+    const [gifSearchInput, setGifSearchInput] = useState("");
+
+    const findGif = (event: { target: { value: SetStateAction<string> } }) => {
+        setGifSearchInput(event.target.value);
+        setGifsLoading(true);
+    };
+
+    const getGifUrl = (fullGifUrl: string) => {
+        // image url which should be passed into chat box
+        const gifObject = {
+            content: fullGifUrl,
+            isImage: true
+        };
+        console.log("gif object", gifObject);
+    };
+
+    const renderGifs = () => (
+        <>
+            <GifWidgetSearch
+                type="text"
+                value={gifSearchInput}
+                onChange={findGif}
+                placeholder="find gif"
+            />
+            <GifList>
+                {!gifsLoading &&
+                    gifList.map((item, index) => (
+                        <GifListImage
+                            key={item.images.preview_gif.url}
+                            src={item.images.preview_gif.url}
+                            onClick={() =>
+                                getGifUrl(item.images.downsized_large.url)
+                            }
+                        />
+                    ))}
+            </GifList>
+        </>
+    );
+
+    useEffect(() => {
+        const getGifsList = async () => {
```

Comment:
For me there should be error handling. Because when no response (giphy is down or w/e) the `response.data.data` will be probably null or undefined or error ocures because we want to acces to a property that doesn't exist. And there is expected type to be IGifWidget[].

Reply:
He should create error handling for all requests, but he know about it ;)